### PR TITLE
Change placement of DebuggerNonUserCodeAttribute

### PR DESCRIPTION
**Pull request by jskeet**
See #1671.

### CODE REVIEW

**Comment by jskeet** [comment]
Reassigned to @pherl is @jtattermusch is away. The Travis build failures are the ones afflicting the whole project, not just this PR.

**Comment by pherl** [comment]
I added @anandolee to take a look.

File: src/google/protobuf/compiler/csharp/csharp_source_generator_base.cc [hunk 1]
```diff
@@ -56,6 +56,10 @@ SourceGeneratorBase::~SourceGeneratorBase() {
 }
 
 void SourceGeneratorBase::WriteGeneratedCodeAttributes(io::Printer* printer) {
+  printer->Print("[global::System.Diagnostics.DebuggerNonUserCodeAttribute]\n");
+}
+
+void SourceGeneratorBase::WriteGeneratedTypeAttributes(io::Printer* printer) {
```
Comment (anandolee) [inline src/google/protobuf/compiler/csharp/csharp_source_generator_base.cc]: It is a waste to have this empty method for future use only.
Let's add this when really needed.

Comment (jskeet) [inline src/google/protobuf/compiler/csharp/csharp_source_generator_base.cc]: My reasoning for keeping it was that we've already done the work to make sure it's called at all the right places. But I don't mind removing it - let's just make sure we keep that removal as a separate commit which is easy to revert. Will add that commit to this PR.

**Comment by anandolee** [comment]
LGTM

**Comment by jskeet** [comment]
Thanks. Will look into the CI failures, and submit if they're unrelated.
